framework/src/util/Utils.cpp

```diff
@@ -154,6 +154,13 @@ std::string GetCurrentWorkingDirectory()
   return std::string();
 }
 
+static const std::string s_CurrWorkingDir = InitCurrentWorkingDirectory();
```

Comment:
Sorry. I finally understand your point. Yes, the Win32 API ```GetCurrentDirectory``` reads a global variable and yes, the only way to guarantee that CppMicroServices never calls into ```GetCurrentDirectory``` concurrently is to call it during static initialization or not at all.

I'm wondering whether there is an even simpler approach now...
Considering how the current working directory is being used in CppMicroServices to construct absolute paths to folders, what about doing away with calling ```GetCurrentDirectory``` altogether and use the Win32 API ```SearchPath``` instead to construct an absolute path?